### PR TITLE
#88: add cookie-message element

### PR DESCRIPTION
**Pull request by talalus**
#88

### CODE REVIEW

File: components/02-elements/cookie-message/CookieMessage.vue [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+  },
+  methods: {
+    closeBar() {
+      const focusable = document.querySelectorAll('button:not([disabled]), a[href], area[href] input:not([disabled]), select:not([disabled]), textarea:not([disabled]), *[tabindex]:not([tabindex="-1"]), object, embed, *[contenteditable]');
+
+      this.openClass = ''
+      localStorage.setItem(this.dataType, 'closed');
+      // after closing message box move focus on first focusable element on the page
+      if (focusable.length) {
+        let firstFocusable = [...focusable].find(el => {
```
Comment (jszczech) [inline components/02-elements/cookie-message/CookieMessage.vue]:
```suggestion
        let firstFocusable = [...focusable].find(el => el.offsetParent !== null)
```

File: components/02-elements/cookie-message/CookieMessage.vue [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        })
+        firstFocusable.focus();
+      }
+    }
+  }
+}
+</script>
+
+<style lang="scss">
+$cookie-message__width: 100% !default;
```
Comment (jszczech) [inline components/02-elements/cookie-message/CookieMessage.vue]: Formatting

File: components/02-elements/cookie-message/CookieMessage.vue [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    },
+    closeAriaLabel: {
+      type: String,
+      required: false,
+      default: ''
+    }
+  },
+  data: () => {
+    return {
+      openClass: ''
```
Comment (jszczech) [inline components/02-elements/cookie-message/CookieMessage.vue]: Maybe more readable would be to have `isOpen: false`, and setting the value to true if it is open. 
In line 3: `:class="['cookie-message', isOpen && 'cookie-message--open']"` - then all classes stay in template. What do you think?

File: components/02-elements/cookie-message/CookieMessage.vue [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  }
+}
+</script>
+
+<style lang="scss">
+$cookie-message__width: 100% !default;
+$cookie-message__background-color: $gray-lightest !default;
+$cookie-message__font-size: $font-size-small !default;
+$cookie-message__font-family: $font-family-base !default;
+$cookie-message__text-color: $gray !default;
```
Comment (jszczech) [inline components/02-elements/cookie-message/CookieMessage.vue]: Lines 101, 106 have the same value. Maybe we could use only one variable?

File: components/02-elements/cookie-message/CookieMessage.stories.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import App from '../../01-globals/app/App.vue'
+import AlpacaCookieMessage from './CookieMessage.vue'
+
+storiesOf('Elements/CookieMessage', module)
+  .add('Default', () => ({
+    components: { App, AlpacaCookieMessage },
+    template: `
+      <app>
+        <alpaca-cookie-message
+          ariaLabel="Cookie policy message"
```
Comment (szafran89) [inline components/02-elements/cookie-message/CookieMessage.stories.js]: aria-label shouldn't be required, please remove

File: components/02-elements/cookie-message/CookieMessage.stories.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+          text="<a href='#' class='cookie-message__link'>Example link</a> Find out more about their purpose and settings in your browser. By browsing the site you are agreeing to use cookies according to your browser settings."
+          closeIcon="close"
+          closeAriaLabel="Close cookie message"
```
Comment (szafran89) [inline components/02-elements/cookie-message/CookieMessage.stories.js]: shouldn't be required

File: components/02-elements/cookie-message/CookieMessage.vue [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+<script>
+  import Icon from '../../01-globals/icon/Icon'
+
+  export default {
+    components: { Icon },
+    props: {
+      ariaLabel: {
+        type: String,
+        required: true
```
Comment (szafran89) [inline components/02-elements/cookie-message/CookieMessage.vue]: set default value

Comment (szafran89) [reply components/02-elements/cookie-message/CookieMessage.vue]: and not required

File: components/02-elements/cookie-message/CookieMessage.vue [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+      },
+      closeLabel: {
+        type: String,
+        required: false,
+        default: null
+      },
+      closeAriaLabel: {
+        type: String,
+        required: false,
+        default: null
```
Comment (szafran89) [inline components/02-elements/cookie-message/CookieMessage.vue]: set default `Close cookie message`

File: components/02-elements/cookie-message/CookieMessage.stories.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
+import {storiesOf} from '@storybook/vue'
```
Comment (szafran89) [inline components/02-elements/cookie-message/CookieMessage.stories.js]: soaces `{ storiesOf }`

File: components/02-elements/cookie-message/CookieMessage.vue [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+    :data-type="dataType"
+    :aria-label="ariaLabel"
+    tabindex="0"
+  >
+    <div class="cookie-message__container">
+      <span
+        class="cookie-message__text"
+        v-html="text"
+      />
+      <button
```
Comment (szafran89) [inline components/02-elements/cookie-message/CookieMessage.vue]: can we use `alpaca-button`

**Pull request merged**